tests/automatic_calibration/test_calibration_controller.py

```diff
@@ -0,0 +1,849 @@
+import itertools
+from datetime import datetime
+from unittest.mock import MagicMock, call, patch
+
+import networkx as nx
+import pytest
+
+from qililab.automatic_calibration import CalibrationController, CalibrationNode
+from qililab.data_management import build_platform
+from qililab.platform.platform import Platform
+
+# flake8: noqa
+#################################################################################
+#################################### SET UPS ####################################
+#################################################################################
+
+path_runcard = "examples/runcards/galadriel.yml"
+
+
+def dummy_comparison_model(obtained: dict, comparison: dict) -> float:
+    """Basic comparison model for testing."""
+    return abs(sum(obtained["y"]) - sum(comparison["y"]))
+
+
+######################
+### NODES CREATION ###
+######################
+zeroth = CalibrationNode(
```

Comment:
Hmm, we'll think about it, but yeah it would reduce the code for sure!

Reply:
We are gonna take your approach for this one, in the end, nice thinking! 🙌

Reply:
Even though we liked your suggestion much, we are not gonna do this in the end, because we are passing 3 different thresholds, and 2 qubit numbers, for getting different values later in the tests!